aiopvapi/hub.py

```diff
@@ -61,13 +82,36 @@ def __eq__(self, other):
 
 
 class Hub(ApiBase):
```

Comment:
WOuld it not be better to have a V2Hub and a V3Hub subclass, then we dont need all the switching

Reply:
I thought about that, not sure.  I thought about leaving UserData and Hub and calling V3 Gateway since that is what Hunter Douglas seems to have moved to.  UserData won't be used in V3 regardless.

Probably should have asked this question prior to doing it as I had the same question myself, should I refactor it to be separate objects for each revision?

Reply:
I dont know the answer, just providing constructive PR feedback :)

Reply:
I'm not sure here either to be honest - we could go subclassing, but then we ask the question re subclassing shades etc too

I have rewritten this one though, So my (unsubmitted) code has done away with the userdata class and makes all the calls necessary under ``query_firmware_v2`` in the same manner ``query_firmware_v3`` does